crates/bitwarden-crypto/src/store/key_rotation.rs

```diff
@@ -45,6 +49,113 @@ pub fn dangerous_get_v2_rotated_account_keys<Ids: KeyIds>(
     })
 }
 
+/// A set of keys where a given `EncryptionKey` is protected by an encrypted public/private
+/// key-pair. The `EncryptionKey` is used to encrypt/decrypt data, while the public/private key-pair
+/// is used to rotate the `EncryptionKey`.
+///
+/// The `PrivateKey` is protected by an `ExternalKey`, such as a `DeviceKey`, or `PrfKey`,
+/// and the `PublicKey` is protected by the `EncryptionKey`. This setup allows:
+///
+///   - Access to `EncryptionKey` by knowing the `ExternalKey`
+///   - Rotation to a `NewEncryptionKey` by knowing the current `UserKey`, without needing access to
+///     the `ExternalKey`
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
+#[cfg_attr(feature = "wasm", derive(Tsify), tsify(into_wasm_abi, from_wasm_abi))]
+pub struct RotateableKeySet {
+    /// `EncryptionKey` protected by encapsulation key
+    encapsulated_encryption_key: UnsignedSharedKey,
+    /// Encapsulation key protected by `EncryptionKey`
+    encrypted_encapsulation_key: EncString,
+    /// Decapsulation key protected by `ExternalKey`
+    wrapped_decapsulation_key: EncString,
+}
+
+impl RotateableKeySet {
+    /// Create a set of keys to allow access to the user key via the provided
+    /// symmetric wrapping key while allowing the user key to be rotated.
+    pub fn new<Ids: KeyIds>(
+        ctx: &KeyStoreContext<Ids>,
+        wrapping_key: &SymmetricCryptoKey,
+        key_to_wrap: Ids::Symmetric,
+    ) -> Result<Self, CryptoError> {
+        let key_pair = AsymmetricCryptoKey::make(crate::PublicKeyEncryptionAlgorithm::RsaOaepSha1);
+
+        #[allow(deprecated)]
+        let key_to_wrap_instance = ctx.dangerous_get_symmetric_key(key_to_wrap)?;
+        // encapsulate encryption key
+        let encapsulated_encryption_key = UnsignedSharedKey::encapsulate_key_unsigned(
+            key_to_wrap_instance,
+            &key_pair.to_public_key(),
+        )?;
+
+        // wrap decapsulation key
+        let wrapped_decapsulation_key = key_pair.to_der()?.encrypt_with_key(wrapping_key)?;
+
+        // wrap encapsulation key with encryption key
+        // Note: Usually, a public key is - by definition - public, so this should not be necessary.
+        // The specific use-case for this function is to enable rotateable key sets, where
+        // the "public key" is not public, with the intent of preventing the server from being able
+        // to overwrite the user key unlocked by the rotateable keyset.
+        let encrypted_encapsulation_key = key_pair
+            .to_public_key()
+            .to_der()?
+            .encrypt_with_key(&key_to_wrap_instance)?;
+
+        Ok(RotateableKeySet {
+            encapsulated_encryption_key,
+            encrypted_encapsulation_key,
+            wrapped_decapsulation_key,
+        })
+    }
+
+    fn unlock<Ids: KeyIds>(
```

Comment:
This method is currently unused since we don't yet have plans to use the rotateable key set on mobile, only to create them. However, it implements functionality that is [currently done in TypeScript](https://github.com/bitwarden/clients/blob/main/libs/auth/src/common/login-strategies/webauthn-login.strategy.ts#L84-L94) that could be moved to the SDK.

I can remove it until we decide to either migrate the browser apps to this SDK method or start using it in the mobile apps.